### ISSUE TITLE
Helm chart currently doesn't expose things needed to specify ingress controllers IPFamilyPolicy & related fields

### ISSUE THREAD

**Issue by samip5** [cilium/cilium#27003]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I was trying to specify needed things for the service, but they're not exposed via helm.
Eg. ipFamilyPolicy nor ipFamilies.

### Cilium Version

cilium-cli: v0.12.1 compiled with go1.18.5 on linux/amd64
cilium image (default): v1.12.0
cilium image (stable): v1.13.4
cilium image (running): v1.14.0-rc.1

### Kernel Version

6.1.0-10-arm64

### Kubernetes Version

v1.27.3+k3s1

### Sysdump

I don't think relevant

### Anything else?

ingress-nginx allows these to be specified as well as externalTrafficPolicy.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
Thanks for this @samip5.

It sounds like you'd like to be able to configure which IP families should be used on the generated Loadbalancer Service used by the Ingress controller?

We're focussing more on the Gateway  API support at the moment, but Gateway API doesn't have a similar directive yet.

I haven't used the IPFamily config on ingress-nginx before, it seems like that's an installation-wide setting. Is that what you're looking for?

(I'm not sure how we'll solve this yet, just wanting to clarify the request).

**Comment by samip5** [comment]
> It sounds like you'd like to be able to configure which IP families should be used on the generated Loadbalancer Service used by the Ingress controller?

Correct, as by default currently that service will only be IPv4 if the cluster's default one is v4 and I would like to not have to manually edit the generated lb so that it would have both. It would be really trivial to add support in the helm chart.

The funny thing actually is that if the cluster's default family is IPv6, it will actually be that instead only.

 (Default family is determined by which of the prefixes is specified first on cluster service cidr and cluster cidr)

**Comment by youngnick** [comment]
Thanks for that, I think we're looking at spending some time on LB Service customisation in this cycle, so I'll see if we can include this one in there as well.

**Comment by samip5** [comment]
I don't think the `need-more-info` is valid anymore here?

**Comment by youngnick** [comment]
Yes, thanks for the ping.

**Comment by samip5** [comment]
Shouldn't be stale.

**Comment by samip5** [comment]
@youngnick Probably this needs a not stale thing?

**Comment by youngnick** [comment]
Yes, thanks @samip5. I'll pin this.
